### DISCUSSION TITLE
Do you generate some of these docs?

### DISCUSSION THREAD

**Discussion by elijahmanor** [LazyVim/LazyVim#84] [General]
Are the new docs (for example those for [keymaps](https://lazyvim.github.io/keymaps)) generated from this repo somehow or are they manually updated in the [lazyvim.github.io](https://github.com/LazyVim/lazyvim.github.io) repo?

**Comment by folke** [comment]
Pretty much all the docs are auto-generated based on files from `LazyVim/LazyVim` and `LazyVim/starter`.

You can check my pretty ugly code for that [here](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/util/build.lua) 

This is only used to generate the docs, all my other code is normally much cleaner 😄 

I use the lua treesitter parser to extract specs and opts for example. 

Then there's some other hacks I use to extract the keymaps for the docs.

A bit messy, but works really well! :)

**Reply by elijahmanor** [reply]
I had a hunch you were automating this, but I wasn't sure where to find it. Nice! I wondered because I noticed a couple of the keymaps I thought could be cleaner in the docs, but I didn't want to make a PR to them if they were generated.

For example, the split window right keymaps have an extra `\` before the pipe `|` character.

![Screenshot 2023-01-20 at 10 35 54 AM](https://user-images.githubusercontent.com/86454/213753337-931a849c-b71b-49f4-88cd-dc7cdaebce90.png)

The source code that is parsed is the following...

```
vim.keymap.set("n", "<leader>w-", "<C-W>s", { desc = "Split window below" })
vim.keymap.set("n", "<leader>w|", "<C-W>v", { desc = "Split window right" })
vim.keymap.set("n", "<leader>-", "<C-W>s", { desc = "Split window below" })
vim.keymap.set("n", "<leader>|", "<C-W>v", { desc = "Split window right" })
```

So I'm guessing the issue is on this line... https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/util/build.lua#L76

**Reply by folke** [reply]
Good catch! That was needed for GitHub flavoured markdown, but it seems it's no longer needed with MDX. Will look into it tomorrow

**Reply by folke** [reply]
Looks like this is an issue with MDX 1 (https://github.com/mdx-js/mdx/issues/838). It's fixed in MDX 2.x, but Docusaurus is still using v1.

I was able to work-around it by using `<code>` blocks instead.

**Reply by elijahmanor** [reply]
Awesome, love the sections for the keymaps as well. Nice fix and addition

**Comment by folke** [comment]
At some point I may cleanup all that code and let users generate docs for their own config, with their custom keymaps, plugin specs, etc...

**Reply by elijahmanor** [reply]
That would be a cool feature, but yeah... one that can come much later.
